controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -635,7 +635,7 @@ private String getTransactionName(String scope, String stream, long segmentId, U
                 });
     }
 
-    public CompletableFuture<WireCommands.SegmentRead> readSegment(String qualifiedName, int offset, int length,
+    public CompletableFuture<WireCommands.SegmentRead> readSegment(String qualifiedName, long offset, int length,
```

Comment:
Please change `SegmentHelperTest.testReadSegment()` to pass long offset values instead of integers to this method.

Reply:
Also in class [ReadSegmentRangeCommand](https://github.com/pravega/pravega/blob/3a11b6d5496bc78c93b0ccaac14e9f6327c9e5bb/cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/ReadSegmentRangeCommand.java#L117) we seem to be type casting the `currentOffset` to "int", which can now be removed.

Reply:
Changes done